UnityProject/Assets/Scripts/Core/Admin/Logs/LogEntry.cs

```diff
@@ -1,35 +1,152 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
 
 namespace Core.Admin.Logs
 {
 	public class LogEntry
 	{
 		public DateTime LogTime { get; } = DateTime.UtcNow;
-		public string Log;
-		public List<AdminActionToTake> AdminActions = new List<AdminActionToTake>();
+		public LogInfo[] Log;
+		public List<AdminActionToTake> AdminActions = null;
 		public Severity LogImportance;
-		public string Perpetrator;
 		public LogCategory Category;
 	}
 
-	public class HumanLogEntry
+	public static class LogUtilities
+	{
+
+		public static LogInfo GetLogInfo(this GameObject go)
+		{
+
+			var LogInfo = new LogInfo();
+
+			var StoredIn = go.GetRootGameObject();
+
+			PlayerInfo WasControlledByPlayer = null;
+			if (StoredIn != null)
+			{
+				WasControlledByPlayer = StoredIn.GetComponentCustom<PlayerScript>()?.PlayerInfo;
+			}
+			else
+			{
+				WasControlledByPlayer = go.Player();
+			}
+
+			if (go == StoredIn)
+			{
+				StoredIn = null;
+			}
+
+
+
+			LogInfo.CoreObject = go;
+			LogInfo.CoreObjectName = go?.name;
+			LogInfo.WasControlledByPlayer = WasControlledByPlayer;
+			LogInfo.WasStoredInObject = StoredIn;
+			LogInfo.WasStoredInObjectName = StoredIn?.name;
+			LogInfo.Info = "";
+
+			if (go == null)
+			{
+				LogInfo.WasAtPositionWorld = Vector3.zero;
+			}
+			else
+			{
+				LogInfo.WasAtPositionWorld = go.AssumedWorldPosServer();
+			}
+
+
+
+			return LogInfo;
+		}
+
+		public static LogInfo GetLogInfo(this string Info)
+		{
+			var LogInfo = new LogInfo();
+			LogInfo.Info = Info;
+			return LogInfo;
+		}
+
+	}
+
+	public enum LogMarker
+	{
+		Core,
+		StoredIn,
+		ControlledBy,
+		Position,
+		Info
+	}
+
+	public struct LogInfo
+	{
+		public GameObject CoreObject;
+		public string CoreObjectName;
+		public GameObject WasStoredInObject;
+		public string WasStoredInObjectName;
+		public PlayerInfo WasControlledByPlayer;
+		public Vector3 WasAtPositionWorld;
+		public string Info;
+
+		public LongTermLogEntry.LogItems SerialiseVersion()
+		{
+			return new LongTermLogEntry.LogItems()
+			{
+				Object = CoreObject.NetIdCommonComponents(),
+				ObjectName = CoreObjectName,
+				StoredIn = WasStoredInObject.NetIdCommonComponents(),
+				StoredInName = WasStoredInObjectName,
+				PlayerAccountID = WasControlledByPlayer?.AccountId,
+				PositionWorld = WasAtPositionWorld.ToSerialiseString(),
+				Info = Info
+			};
+		}
+
+	}
+
+	public struct LongTermLogEntry
```

Comment:
This is still a bad name for it.
Just call it "SerializedLongEntry" or "StoredLongEntry"